src/prefect/serialization/state.py

```diff
@@ -32,14 +32,19 @@ class BaseStateSchema(ObjectSchema):
     class Meta:
         object_class = state.State
 
+    context = fields.Dict(
+        key=fields.Str(), values=fields.List(fields.Str()), allow_none=True
```

Comment:
I'm currently making this serializer hyper-specific.  I could relax this if necessary.

Reply:
Actually now that I've said this out loud I want to change it in this PR brb

Reply:
Updated to allow for generic JSON values in the dict.